projects/storefrontlib/src/lib/ui/pages/product-page/current-product.service.ts

```diff
@@ -0,0 +1,20 @@
+import { Injectable } from '@angular/core';
+import { filter, map, switchMap } from 'rxjs/operators';
+import { Observable } from 'rxjs';
+import { Product, ProductService, RoutingService } from '@spartacus/core';
+
+@Injectable()
+export class CurrentProductService {
+  constructor(
+    private routingService: RoutingService,
+    private productService: ProductService
+  ) {}
+
+  getProduct(): Observable<Product> {
+    return this.routingService.getRouterState().pipe(
+      map(state => state.state.params['productCode']),
+      filter(Boolean),
```

Comment:
I would suggest to change this filter into something like `.filter(a => !!a)` instead (easier to understand) - but up to you

Reply:
additionally a small disadvantage of the current filter: https://github.com/Reactive-Extensions/RxJS/issues/948#issuecomment-279696051

Reply:
updated